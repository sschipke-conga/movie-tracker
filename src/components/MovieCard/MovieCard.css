.movie {
  /* height: 500px; */
  margin: 10px;
}

.movie h2 {
  height: 60px;
}
.card_image {
  height: 350px;
  min-height: 300px;
  max-width: 250px;
}