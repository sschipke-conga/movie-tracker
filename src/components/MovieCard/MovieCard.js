import React from 'react';
import './MovieCard.css';
import star from '../../images/bluestar.svg';
import favstar from '../../images/fav-star.svg'

const MovieCard = ({movie}) => {

  return (
    <div className='movie'>
      <img alt='star indicating is a movie is favorited or not' src={star} className='card__image-favorite' />
      <img className='card__image' alt="movie poster" src={`https://image.tmdb.org/t/p/original/${movie.poster_path}`} />
      <div className='card__header'>
        <p>
          {movie.title}
        </p>
      </div>
      <h4 className='card__release-date'>
        {movie.release_date.slice(0,4)}
      </h4>
    </div>
  )
}

export default MovieCard;