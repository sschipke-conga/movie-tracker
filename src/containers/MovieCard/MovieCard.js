import React from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';
import { connect } from 'react-redux';
import './MovieCard.css';
import star from '../../images/star.svg';
import favstar from '../../images/fav-star.svg';

export const MovieCard = ({movie, toggleFavorites, favorites}) => {
  let isFavorite = favorites.map(movie => movie.title).includes(movie.title);
  let faveImg = isFavorite ? favstar : star;
  let faveClass = isFavorite ? 'favorite' : '';
  const id = movie.movie_id;
  return (
    <Link to={`/movie/${id}`}>
      <div className={`movie ${faveClass}`}  movie_id={movie.movie_id}>
        <img alt='star indicating is a movie is favorited or not' 
        src={faveImg} 
        className='card__image-favorite' 
          onClick={ (e) => {
            toggleFavorites(e, movie)
          }}
        />
        <img 
        className='card__image' 
        alt="movie poster" 
        src={`https://image.tmdb.org/t/p/original/${movie.poster_path}`} 
        />
        <div className='card__header'>
          <p>
            {movie.title}
          </p>
        </div>
        <h4 className='card__release-date'>
          {movie.release_date.slice(0,4)}
        </h4>
      </div>
    </Link>
  );
}

export const mapStateToProps = state => ({
  favorites: state.favorites
});

export default connect(mapStateToProps)(MovieCard);

MovieCard.propTypes = {
  movie: PropTypes.object.isRequired,
  toggleFavorites: PropTypes.func.isRequired,
  favorites: PropTypes.array.isRequired
}