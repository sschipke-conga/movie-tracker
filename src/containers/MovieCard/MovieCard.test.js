import React from 'react';
import { shallow } from 'enzyme';
import { MovieCard, mapStateToProps } from './MovieCard';

describe('MovieCard',() => {
  let wrapper, mockMovie, mockFavorites;
  const mockToggleFavorites = jest.fn()
  beforeEach(() => {
    mockMovie = {
      "poster_path": "/vqzNJRH4YyquRiWxCCOH0aXggHI.jpg",
      "id": 290859,
      "title": "Terminator: Dark Fate",
      "vote_average": 7.4,
      "overview": "More than two decades have passed since Sarah Connor prevented Judgment Day, changed the future, and re-wrote the fate of the human race. Dani Ramos is living a simple life in Mexico City with her brother and father when a highly advanced and deadly new Terminator – a Rev-9 – travels back through time to hunt and kill her. Dani's survival depends on her joining forces with two warriors: Grace, an enhanced super-soldier from the future, and a battle-hardened Sarah Connor. As the Rev-9 ruthlessly destroys everything and everyone in its path on the hunt for Dani, the three are led to a T-800 from Sarah’s past that may be their last best hope.",
      "release_date": "2019-11-01"
    }
    mockFavorites = [{
      "poster_path": "/vqzNJRH4YyquRiWxCCOH0aXggHI.jpg",
      "id": 290859,
      "title": "Terminator: Dark Fate",
      "vote_average": 7.4,
      "overview": "More than two decades have passed since Sarah Connor prevented Judgment Day, changed the future, and re-wrote the fate of the human race. Dani Ramos is living a simple life in Mexico City with her brother and father when a highly advanced and deadly new Terminator – a Rev-9 – travels back through time to hunt and kill her. Dani's survival depends on her joining forces with two warriors: Grace, an enhanced super-soldier from the future, and a battle-hardened Sarah Connor. As the Rev-9 ruthlessly destroys everything and everyone in its path on the hunt for Dani, the three are led to a T-800 from Sarah’s past that may be their last best hope.",
      "release_date": "2019-11-01"
    }]
    wrapper = shallow(<MovieCard 
      movie = {mockMovie}
      favorites = {mockFavorites}
      toggleFavorites = {mockToggleFavorites}
      />)
  });

  it('should match the snapshot with all the data passed in correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should call toggleFavorites when the star is clicked',() => {
    const event = { preventDefault:jest.fn() }
    wrapper.find('.card__image-favorite').simulate('click', event);
    expect(mockToggleFavorites).toHaveBeenCalledWith(event, mockMovie)
  });

})

describe('MovieCard Redux',() => {

  it('should map favorite array to state',() => {
    const mockFavorites = [
      {
        "poster_path": "/udDclJoHjfjb8Ekgsd4FDteOkCU.jpg",
        "id": 475557,
        "title": "Joker",
        "vote_average": 8.6,
        "overview": "During the 1980s, a failed stand-up comedian is driven insane and turns to a life of crime and chaos in Gotham City while becoming an infamous psychopathic crime figure.",
        "release_date": "2019-10-04"
      },
      {
        "poster_path": "/tBuabjEqxzoUBHfbyNbd8ulgy5j.jpg",
        "id": 420809,
        "title": "Maleficent: Mistress of Evil",
        "vote_average": 7.1,
        "overview": "Maleficent and her goddaughter Aurora begin to question the complex family ties that bind them as they are pulled in different directions by impending nuptials, unexpected allies, and dark new forces at play.",
        "release_date": "2019-10-18"
      }
    ];
    const mockState = {
      favorites:mockFavorites
    };

    const expectedState = { 
      favorites:mockState.favorites
    }
    const mappedProps = mapStateToProps(mockState)
    expect(mappedProps).toEqual(expectedState)
  })

})
