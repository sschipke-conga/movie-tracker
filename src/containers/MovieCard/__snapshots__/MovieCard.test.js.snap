// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MovieCard should match the snapshot with all the data passed in correctly 1`] = `
<Link
  to="/movie/undefined"
>
  <div
    className="movie favorite"
  >
    <img
      alt="star indicating is a movie is favorited or not"
      className="card__image-favorite"
      onClick={[Function]}
      src="fav-star.svg"
    />
    <img
      alt="movie poster"
      className="card__image"
      src="https://image.tmdb.org/t/p/original//vqzNJRH4YyquRiWxCCOH0aXggHI.jpg"
    />
    <div
      className="card__header"
    >
      <p>
        Terminator: Dark Fate
      </p>
    </div>
    <h4
      className="card__release-date"
    >
      2019
    </h4>
  </div>
</Link>
`;
